legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/domain/DomainParseTreeWalker.java

```diff
@@ -395,6 +397,10 @@ private org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain
         func.returnType = ctx.functionTypeSignature().type().getText();
         func.returnMultiplicity = this.buildMultiplicity(ctx.functionTypeSignature().multiplicity().multiplicityArgument());
         func.sourceInformation = this.walkerSourceInformation.getSourceInformation(ctx);
+        String functionSignature =   LazyIterate.collect(func.parameters, HelperValueSpecificationGrammarComposer::getParameterSignature).select(Objects::nonNull).makeString("__")
```

Comment:
Is this change needed because when going from text to pmcd we can't have 2 elements with the same package and name?
Do you know where that constraint comes from?
My thinking is that the grammar/protocol should not have to handle this. 
It is when we compile (protocol -> metamodel)  the function we need to provide and generate the full function name.